lib/tag.rb

```diff
@@ -0,0 +1,11 @@
+require_relative 'dom_parser.rb'
+
+class Tag < Struct.new(:text, :type, :classes, :id, :name, :children, :parent, :closing, :depth)
```

Comment:
I would recommend wrapping all data in one container (hash or array) because of the varying nature of html attributes